pytorch_lightning/overrides/distributed.py

```diff
@@ -42,11 +43,11 @@ def __init__(self, pl_module: "pl.LightningModule") -> None:
         super().__init__(pl_module)
 
 
-def _find_tensors(obj):  # pragma: no-cover
-    r"""
-    Recursively find all tensors contained in the specified object.
-    """
-    if isinstance(obj, torch.Tensor):
+def _find_tensors(
+    obj: Union[Tensor, list, tuple, dict, Any]
+) -> Union[List[Tensor], itertools.chain]:  # pragma: no-cover
```

Comment:
IMO this function could wrap with list the `itertools.chains` instead of having the callers do it.

Reply:
maybe the implementation is this way to avoid a list conversion in the recursive calls

Reply:
It could provide a base recursive case that does it.

But it's fine, it's like this in torch itself. Resolving